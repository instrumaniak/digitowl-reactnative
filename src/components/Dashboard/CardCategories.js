/**
 *   Dashboard: Categories section
 */

import React, { PureComponent } from 'react'
import { View, StyleSheet } from 'react-native'
import { PieChart } from 'react-native-svg-charts'
import { Circle, G, Line, Text as SText } from 'react-native-svg'

import {
  Card,
  CardItem,
  Text
} from 'native-base'

import { human } from 'react-native-typography'
import { colors } from '../../config'
import CategoryItem from './CategoryItem'

const data2 = [ 50, 53, 24, 5, 20, 80 ]

const pieData = data2
  .filter(value => value > 0)
  .map((value, index) => ({
    value,
    svg: {
      fill: colors.grey1
    },
    key: `pie-${index}`
  }))

const PieLabels = ({ slices }) => {
  return slices.map((slice, index) => {
    const { labelCentroid, pieCentroid, data } = slice
    return (
      <G key={index}>
        <Line
          x1={ labelCentroid[0] }
          y1={ labelCentroid[1] }
          x2={ pieCentroid[0] }
          y2={ pieCentroid[1] }
          stroke={ data.svg.fill }
        />
        <Circle
          cx={ labelCentroid[0] }
          cy={ labelCentroid[1] }
          r={ 20 }
          fill={ colors.grey }
        />
        <SText
          x={ labelCentroid[0] - 7 }
          y={ labelCentroid[1] + 5 }
          style={{ ...human.caption2Object }}
          stroke={ colors.grey2 }
        >
          { data.value} %
        </SText>
      </G>
    )
  })
}

class CardCategories extends PureComponent {
  render() {
    return (
      <Card style={{ elevation: 2 }}>
        <CardItem>
          <Text style={styles.cardHeaderText}>Categories</Text>
        </CardItem>
        <CardItem>
          <View style={styles.btnContainer}>
            <View style={[styles.overviewButton, styles.oBtnActive]}>
              <Text style={styles.obtnHeading}>Total Expenses</Text>
              <Text style={styles.obtnValue}>- 4 628 USD</Text>
            </View>
            <View style={styles.overviewButton}>
              <Text style={styles.obtnHeading}>Total Income</Text>
              <Text style={styles.obtnValue}>+ 5 884 USD</Text>
            </View>
          </View>
        </CardItem>

      {/* Chart */}
        <CardItem>
          <PieChart
            style={{ flex:1, height: 250 }}
            data={pieData}
            innerRadius={30}
            outerRadius={60}
            labelRadius={90}
          >
            <PieLabels />
          </PieChart>
        </CardItem>

      {/* Category Item list */}
        <CardItem>
          <CategoryItem
            iconText="$"
            title="Food and Drink"
            description="26 transations"
            value="-1 804.90 USD"
          />
        </CardItem>
        <CardItem>
          <CategoryItem
            iconText="$"
            title="Travel"
            description="17 transactions"
            value="-694.20 USD"
          />
        </CardItem>
        <CardItem>
          <CategoryItem
            iconText="$"
            title="Rent"
            description="1 transactions"
            value="-555.40 USD"
          />
        </CardItem>
        <CardItem>
          <CategoryItem
            iconText="$"
            title="Entertainment"
            description="5 transations"
            value="-462.80 USD"
          />
        </CardItem>
        <CardItem>
          <CategoryItem
            iconText="$"
            title="Energy"
            description="6 transactions"
            value="-450.50 USD"
          />
        </CardItem>

        <CardItem>
          <View style={{
            flex: 1,
            alignItems: 'center'
          }}>
            <Text style={{
              ...human.subheadObject,
              color: colors.grey1,
              textDecorationLine: 'underline'
            }}>All Categories (7)</Text>
          </View>
        </CardItem>
      </Card>
    )
  }
}

const styles = StyleSheet.create({
  cardHeaderText: {
    ...human.headlineObject,
    color: colors.grey2
  },
  btnContainer: {
    flex: 1,
    flexDirection: 'row'
  },
  btnLight: {
    backgroundColor: 'white',
    flex: 1,
    marginHorizontal: 5,
    justifyContent: 'center'
  },
  overviewButton: {
    flex: 1,
    alignItems: 'center',
    padding: 5
  },
  oBtnActive: {
    backgroundColor: 'rgba(0,0,0,0.05)',
    borderRadius: 5
  },
  obtnHeading: {
    ...human.bodyObject,
    color: colors.grey1,
    marginBottom: 5
  },
  obtnValue: {
    ...human.title3Object,
    color: colors.grey2
  }
})

export default CardCategories
