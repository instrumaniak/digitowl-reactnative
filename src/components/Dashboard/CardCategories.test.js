/**
 *   Tests for component: CardCategories
 */

import React from 'react'
import CardCategories from './CardCategories'
import renderer from 'react-test-renderer'

describe('CardCategories component', () => {
  it('Renders correctly', () => {
    const tree = renderer.create(<CardCategories />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
