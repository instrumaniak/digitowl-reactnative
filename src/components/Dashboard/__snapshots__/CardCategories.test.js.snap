// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardCategories component Renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#ccc",
        "borderLeftWidth": 0.5,
        "borderRadius": 2,
        "borderRightWidth": 0.5,
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexWrap": "nowrap",
        "marginBottom": 5,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 1.5,
      },
      Object {
        "elevation": 2,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "color": "#5B6267",
          },
        ]
      }
      uppercase={false}
    >
      Categories
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "padding": 5,
            },
            Object {
              "backgroundColor": "rgba(0,0,0,0.05)",
              "borderRadius": 5,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "color": "#5B6267",
                "fontSize": 15,
                "marginBottom": 5,
              },
            ]
          }
          uppercase={false}
        >
          Total Expenses
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "color": "#5B6267",
                "fontSize": 20,
                "fontWeight": "bold",
              },
            ]
          }
          uppercase={false}
        >
          - 4 628 USD
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "padding": 5,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "color": "#5B6267",
                "fontSize": 15,
                "marginBottom": 5,
              },
            ]
          }
          uppercase={false}
        >
          Total Income
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "color": "#5B6267",
                "fontSize": 20,
                "fontWeight": "bold",
              },
            ]
          }
          uppercase={false}
        >
          + 5 884 USD
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "height": 250,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#BBC8CA",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "width": 50,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 35,
            }
          }
        >
          $
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 15,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "#5B6267",
                "fontSize": 20,
              }
            }
          >
            Food and Drink
          </Text>
          <Text
            style={
              Object {
                "color": "#BBC8CA",
                "fontSize": 16,
              }
            }
          >
            26 transations
          </Text>
        </View>
        <View>
          <Text
            style={
              Object {
                "color": "#5B6267",
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            -1 804.90 USD
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#BBC8CA",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "width": 50,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 35,
            }
          }
        >
          $
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 15,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "#5B6267",
                "fontSize": 20,
              }
            }
          >
            Travel
          </Text>
          <Text
            style={
              Object {
                "color": "#BBC8CA",
                "fontSize": 16,
              }
            }
          >
            17 transactions
          </Text>
        </View>
        <View>
          <Text
            style={
              Object {
                "color": "#5B6267",
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            -694.20 USD
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#BBC8CA",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "width": 50,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 35,
            }
          }
        >
          $
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 15,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "#5B6267",
                "fontSize": 20,
              }
            }
          >
            Rent
          </Text>
          <Text
            style={
              Object {
                "color": "#BBC8CA",
                "fontSize": 16,
              }
            }
          >
            1 transactions
          </Text>
        </View>
        <View>
          <Text
            style={
              Object {
                "color": "#5B6267",
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            -555.40 USD
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#BBC8CA",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "width": 50,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 35,
            }
          }
        >
          $
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 15,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "#5B6267",
                "fontSize": 20,
              }
            }
          >
            Entertainment
          </Text>
          <Text
            style={
              Object {
                "color": "#BBC8CA",
                "fontSize": 16,
              }
            }
          >
            5 transations
          </Text>
        </View>
        <View>
          <Text
            style={
              Object {
                "color": "#5B6267",
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            -462.80 USD
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#BBC8CA",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "width": 50,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 35,
            }
          }
        >
          $
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 15,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "#5B6267",
                "fontSize": 20,
              }
            }
          >
            Energy
          </Text>
          <Text
            style={
              Object {
                "color": "#BBC8CA",
                "fontSize": 16,
              }
            }
          >
            6 transactions
          </Text>
        </View>
        <View>
          <Text
            style={
              Object {
                "color": "#5B6267",
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            -450.50 USD
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "#5B6267",
              "textDecorationLine": "underline",
            },
          ]
        }
        uppercase={false}
      >
        All Categories (7)
      </Text>
    </View>
  </View>
</View>
`;
